src/dlr_common.cc

```diff
@@ -0,0 +1,59 @@
+#include "dlr_common.h"
+#include <dmlc/filesystem.h>
+
+
+using namespace dlr;
+
+std::string dlr::GetBasename(const std::string& path) {
+#ifdef _WIN32
+  /* remove any trailing backward or forward slashes
+     (UNIX does this automatically) */
+  std::string path_;
+  std::string::size_type tmp = path.find_last_of("/\\");
+  if (tmp == path.length() - 1) {
+    size_t i = tmp;
+    while ((path[i] == '/' || path[i] == '\\') && i >= 0) {
+      --i;
+    }
+    path_ = path.substr(0, i + 1);
+  } else {
+    path_ = path;
+  }
+  std::vector<char> fname(path_.length() + 1);
+  std::vector<char> ext(path_.length() + 1);
+  _splitpath_s(path_.c_str(), NULL, 0, NULL, 0,
+    &fname[0], path_.length() + 1, &ext[0], path_.length() + 1);
+  return std::string(&fname[0]) + std::string(&ext[0]);
+#else
+  char* path_ = strdup(path.c_str());
+  char* base = basename(path_);
+  std::string ret(base);
+  free(path_);
+  return ret;
+#endif
+}
+
+void dlr::listdir(const std::string& dirname, std::vector<std::string> &paths) {
```

Comment:
Let's use CamelCase for naming to keep consistency.

Reply:
fixed